tests/conftest.py

```diff
@@ -19,6 +19,82 @@ def _test_environment(monkeypatch):
     monkeypatch.setenv("AWS_SESSION_TOKEN", "testing")
 
 
+@pytest.fixture
+def mocked_bucket():
+    bucket_name = "mocked-bucket"
+    with mock_aws():
+        s3 = boto3.client("s3")
+        s3.create_bucket(Bucket=bucket_name)
+        yield bucket_name
+
+
+@pytest.fixture
+def mocked_s3_one_file_per_file_id(mocked_bucket):
+    s3 = boto3.client("s3")
+    s3.put_object(
+        Body=mocked_bucket,
+        Bucket="mocked-bucket",
+        Key="one-to-one/aaaa_001_01.pdf",
+    )
+    s3.put_object(
+        Body=mocked_bucket,
+        Bucket="mocked-bucket",
+        Key="one-to-one/aaaa_002_01.pdf",
+    )
+    return mocked_bucket
+
+
+@pytest.fixture
+def mocked_s3_many_files_per_file_id(mocked_bucket):
+    s3 = boto3.client("s3")
+    s3.create_bucket(Bucket=mocked_bucket)
+    s3.put_object(
+        Body="",
+        Bucket=mocked_bucket,
+        Key="many-to-one/bbbb_003_01.pdf",
+    )
+    s3.put_object(
+        Body="",
+        Bucket=mocked_bucket,
+        Key="many-to-one/bbbb_003_02.pdf",
+    )
+    s3.put_object(
+        Body="",
+        Bucket=mocked_bucket,
+        Key="many-to-one/bbbb_004_01.pdf",
+    )
+    s3.put_object(
+        Body="",
+        Bucket=mocked_bucket,
+        Key="many-to-one/bbbb_003_01.jpg",
+    )
+    return mocked_bucket
+
+
+@pytest.fixture
+def mocked_s3_ddc_one_file_per_file_id():
```

Comment:
What's difference between this and `mocked_s3_one_file_per_file_id`? Could we get away with just one?

Reply:
I suppose there isn't. I think the test I'm actually looking for is making sure that `parse_value_from_text()` can successfully get the item identifier from the S3 file object key given the regex specified in the config. 🤔 

Whenever we get a new source, I think we'll need to do two things:
* Add a source config JSON to `config/`
* Add a test to confirm the regex specified in the config successfully retrieves the item identifier

I updated the tests:
* [Verify regex for DDC works ](https://github.com/MITLibraries/dspace-api-python-scripts/pull/38/files#diff-08a6944653efc734f4c8cb856fc854b9f21d655ca69796af2b170c53bff7919eR50-R58):
   * Looking at the files Joe sent, I see two file name formats: 
      a. [4336711c-c826-46cd-a39c-1739450932eb-02-000463360.pdf](https://s3.console.aws.amazon.com/s3/object/cdps-storage-stage-dissemination?region=us-east-1&bucketType=general&prefix=dipstore2/3e4b/7286/1d74/43f4/ac0a/cbd4/a859/f1ba/2024_028RR_002-dc01295e-5248-4770-93aa-9c22b2f2e07c/objects/4336711c-c826-46cd-a39c-1739450932eb-02-000463360.pdf)
      b. [0a3b7ebd-c819-40a5-90ed-f941877c0f4b-02-000065980_005.jpg](https://s3.console.aws.amazon.com/s3/object/cdps-storage-stage-dissemination?region=us-east-1&bucketType=general&prefix=dipstore2/6ffa/30cc/1dd9/4da9/8480/fa26/8cbe/9c05/2023_068RR_001-aa2a33c4-4222-42b8-98d4-c44e5831f3fd/objects/0a3b7ebd-c819-40a5-90ed-f941877c0f4b-02-000065980_005.jpg)

     I had to make a small update to [REGEX_ID_DDC](https://github.com/MITLibraries/dspace-api-python-scripts/pull/38/files#diff-08a6944653efc734f4c8cb856fc854b9f21d655ca69796af2b170c53bff7919eR23) to exclude everything after the underscore from file names formatted like **b.**

* [Add a test to verify the `bitstream_folders` parameter works](https://github.com/MITLibraries/dspace-api-python-scripts/pull/38/files#diff-08a6944653efc734f4c8cb856fc854b9f21d655ca69796af2b170c53bff7919eR115-R125)
   **Note**: A [mocked bucket fixture](https://github.com/MITLibraries/dspace-api-python-scripts/pull/38/files#diff-e52e4ddd58b7ef887ab03c04116e676f6280b824ab7469d5d3080e5cba4f2128R58-R68) was created for this test. Currently the config JSON is specific to (a) source and (b) S3 bucket structure. Might be nice update this in the future, but I think it is beyond the scope of this PR. Let me know what you think!